File: backend/src/main/java/com/gistacsl/mosaic/robot/enumerate/RobotStatus.java
package com.gistacsl.mosaic.robot.enumerate;

import com.gistacsl.mosaic.common.enumerate.SerializableEnum;

public enum RobotStatus implements SerializableEnum {
    READY_TO_CONNECT(0),
    RTC_CONNECTING(1),
    RTC_CONNECTED(2),
    RTC_DISCONNECTING(3),
    RTC_FAILED(4),
    DISCONNECTED(5),
    WS_CONNECTED(6);

    private final int value;

    RobotStatus(int value) {
        this.value = value;
    }

    @Override
    public int getValue() {
        return this.value;
    }
}

File: frontend/src/utils/robot-status.ts
export const convertRobotStatusToString = (status: number) => {
  if (status === 0) return "READY_TO_CONNECT"
  if (status === 1) return "RTC_CONNECTING"
  if (status === 2) return "RTC_CONNECTED"
  if (status === 3) return "RTC_DISCONNECTING"
  if (status === 4) return "RTC_FAILED"
  if (status === 5) return "DISCONNECTED"
  if (status === 6) return "WS_CONNECTED"
  return "Unknown"
}